### PR TITLE
[issue-644]Update CarbonFile to support Federation

### PR DESCRIPTION
**Pull request by Hexiaoqiao**
#644

### CODE REVIEW

**Comment by Hexiaoqiao** [comment]
retest this please

**Comment by Hexiaoqiao** [comment]
retest this please

File: core/src/main/java/org/carbondata/core/datastorage/store/filesystem/AbstractDFSCarbonFile.java [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+  @Override public String getAbsolutePath() {
+    return fileStatus.getPath().toString();
+  }
+
+  @Override public String getName() {
+    return fileStatus.getPath().getName();
+  }
+
+  @Override public boolean isDirectory() {
+    return fileStatus.isDir();
```
Comment (ravipesala) [inline core/src/main/java/org/carbondata/core/datastorage/store/filesystem/AbstractDFSCarbonFile.java]: `isDir` deprecated so please use `isDirectory`

File: core/src/main/java/org/carbondata/core/datastorage/store/filesystem/ViewFSCarbonFile.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  public CarbonFile getParentFile() {
+    return new ViewFSCarbonFile(fileStatus.getPath().getParent());
+  }
+
+  @Override
+  public boolean renameForce(String changetoName) {
+    FileSystem fs;
+    try {
+      fs = fileStatus.getPath().getFileSystem(FileFactory.getConfiguration());
+      if (fs instanceof ViewFileSystem) {
```
Comment (ravipesala) [inline core/src/main/java/org/carbondata/core/datastorage/store/filesystem/ViewFSCarbonFile.java]: no need to type cast, these operations could be done directly on `FileSystem fs`

File: core/src/main/java/org/carbondata/core/datastorage/store/filesystem/ViewFSCarbonFile.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      files[i] = new ViewFSCarbonFile(listStatus[i]);
+    }
+    return files;
+  }
+
+  @Override
+  public CarbonFile[] listFiles() {
+    FileStatus[] listStatus = null;
+    try {
+      if (null != fileStatus && fileStatus.isDir()) {
```
Comment (ravipesala) [inline core/src/main/java/org/carbondata/core/datastorage/store/filesystem/ViewFSCarbonFile.java]: `isDir` is deprecated

File: core/src/main/java/org/carbondata/core/datastorage/store/impl/AbstractDFSFileHolderImpl.java [hunk 4]
```diff
@@ -0,0 +1,183 @@
+/*
```
Comment (ravipesala) [inline core/src/main/java/org/carbondata/core/datastorage/store/impl/AbstractDFSFileHolderImpl.java]: No need to create abstract class as there is no deviation in code for both HDFS and ViewFS. So please keep the same old `HDFSFileHolderImpl` and use it for both.

File: processing/src/main/java/org/carbondata/processing/util/CarbonDataProcessorUtil.java [hunk 5]
```diff
@@ -125,6 +122,8 @@ public static void renameBadRecordsFromInProgressToNormal(String storeLocation)
     CarbonFile carbonFile = null;
     if (fileType.equals(FileFactory.FileType.HDFS)) {
```
Comment (ravipesala) [inline processing/src/main/java/org/carbondata/processing/util/CarbonDataProcessorUtil.java]: Please directly use `CarbonFile carbonFile = FileFactory.getCarbonFile(badLogStoreLocation, fileType);` . Remove the unnecessary code

**Comment by Hexiaoqiao** [comment]
@ravipesala  thanks for your comments.
i think maybe `AbstractDFSFileHolderImpl`, `HDFSFileHolderImpl ` and `ViewFSFileHolderImpl` should be reserved considering maybe we have some individualization requirement. FYI.

**Comment by Hexiaoqiao** [comment]
@ravipesala your advice is a better choice, i submit a latest pr, pls review.

**Comment by Hexiaoqiao** [comment]
retest this please

**Comment by Hexiaoqiao** [comment]
@ravipesala please continue review.

**Comment by ravipesala** [comment]
LGTM. Thank you for working on this. Merged to master

**Pull request merged**